builtin/game/register.lua

```diff
@@ -398,7 +398,9 @@ local function make_registration_wrap(reg_fn_name, clear_fn_name)
 
 	local orig_clear_fn = core[clear_fn_name]
 	core[clear_fn_name] = function()
-		list = {}
+		for k in pairs(list) do
+		 list[k]=nil
```

Comment:
Indentation is one **tab**.  Also, use spaces around binary operators.  See http://dev.minetest.net/Lua_code_style_guidelines.